lib/azure/armrest/armrest_service.rb

```diff
@@ -102,23 +57,10 @@ def fetch_token
       # the new portal or the New-AzureRoleAssignment powershell command.
       #
       def self.configure(options)
```

Comment:
The old design place the configuration class inside `ArmrestService`, thus make sense to provide this factory method to instantiate a configuration object.

The new design move the configuration class as an independent class. A normal usage will be for a user to directly instantiate a configuration object and pass it to create services. This factory method should remain here only for backward compatible reason. 

However this factory method finds the `subscription_id` if not exist. This makes the factory not obsolete but necessary, which kind of fail the purpose of this refactoring work.